vine/synchronization.py

```diff
@@ -32,6 +32,13 @@ def all_done():
     Note that you cannot add new promises to a barrier after
     the barrier is fulfilled.
     """
+    __slots__ = (
+            'p', 'args', 'kwargs', '_value', 'size',
+            'ready', 'reason', 'cancelled', 'finalized',
+            '__weakref__',
+            # adding '__dict__' to get dynamic assignment
+            "__dict__",
+        )
```

Comment:
Fix indentation (maybe add Black/Ruff to repo?)

```suggestion
    __slots__ = (
        'p', 'args', 'kwargs', '_value', 'size',
        'ready', 'reason', 'cancelled', 'finalized',
        '__weakref__',
        # adding '__dict__' to get dynamic assignment
        "__dict__",
    )
```